src/rai/rai/tools/ros/native.py

```diff
@@ -223,6 +225,57 @@ def _run(self, topic_name: str):
             return str(msg), {}
 
 
+class Ros2GenericServiceCallerInput(BaseModel):
+    service_name: str = Field(..., description="Name of the ROS2 service to call")
+    service_type: str = Field(
+        ..., description="Type of the ROS2 service in typical ros2 format"
+    )
+    request_args: Dict[str, Any] = Field(
+        ..., description="Arguments for the service request"
+    )
+
+
+class Ros2GenericServiceCaller(Ros2BaseTool):
```

Comment:
Please inherit from `Ros2AsyncBaseTool`:
https://github.com/RobotecAI/rai/blob/0333ea219842eeb98d3c41000a8f0b97c37712b0/src/rai/rai/tools/ros/native_actions.py#L58

Also, please register this tool here: https://github.com/RobotecAI/rai/blob/0333ea219842eeb98d3c41000a8f0b97c37712b0/src/rai/rai/node.py#L412

I can see that you spin the tool inside `_run`. If this tool is attached directly to `Rai2BaseNode`, it might cause deadlock / `rclpy` errors, because `RaiBaseNode` is usually being spinned already.